geoopt/manifolds/lorentz/__init__.py

```diff
@@ -0,0 +1,217 @@
+import torch as th
+import torch.nn
+import numpy as np
+from typing import Tuple, Optional
+from . import math
+import geoopt
+from ..base import Manifold, ScalingInfo
+from ...utils import size2shape, broadcast_shapes
+
+__all__ = ["Lorentz"]
+
+_lorentz_ball_doc = r"""
+    Lorentz model
+
+    Parameters
+    ----------
+    k : float|tensor
+        manifold negative curvature
+
+    Notes
+    -----
+    It is extremely recommended to work with this manifold in double precision
+"""
+
+
+class Lorentz(Manifold):
+    __doc__ = r"""{}
+    """
+
+    ndim = 1
+    reversible = False
+    name = "Lorentz"
+    __scaling__ = Manifold.__scaling__.copy()
+
+    def __init__(self, k=1.0, learnable=False):
+        super().__init__()
+        k = torch.as_tensor(k)
+        if not torch.is_floating_point(k):
+            k = k.to(torch.get_default_dtype())
+        self.k = torch.nn.Parameter(k, requires_grad=learnable)
+
+    def _check_point_on_manifold(
+        self, x: torch.Tensor, *, atol=1e-5, rtol=1e-5
+    ) -> Tuple[bool, Optional[str]]:
+        dn = x.size(0)
+        x = x ** 2
+        quad_form = -x[0] + x[1:].sum()
+        ok = torch.allclose(quad_form, -self.k, atol=atol, rtol=rtol)
+        if not ok:
+            reason = f"'x' minkowski quadratic form is not equal to {-self.k.item()}"
```

Comment:
k may be not a scalar

Reply:
I don't see the problem here.

Reply:
I see that now, you mean non constant curvature, ok.